index.js

```diff
@@ -146,18 +149,21 @@ var TruffleContractSchema = {
   // - Resolves as validated `contractObj`
   // - Rejects with list of errors from schema validator
   validate: function(contractObj) {
-    var ajv = new Ajv();
-    var validate = ajv.compile(contractSchema);
-    if (validate(contractObj)) {
+    var ajv = new Ajv({ useDefaults: true });
+    ajv.addSchema(abiSchema);
```

Comment:
For my edification: How does this know that `contractObj.abi` is what should be validated with the abiSchema, vs. the `contractObj` itself?